packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -226,14 +226,61 @@ function WPNonProductLineItem( {
 	/* eslint-enable wpcalypso/jsx-classname-namespace */
 }
 
-function GSuiteUsersList( { product }: { product: ResponseCartProduct } ) {
-	const users = product.extra?.google_apps_users ?? [];
+function EmailMeta( { product, isRenewal }: { product: ResponseCartProduct; isRenewal: boolean } ) {
+	const translate = useTranslate();
+
+	if ( isRenewal ) {
+		let numberOfMailboxes = null;
+
+		if ( isGSuiteOrExtraLicenseProductSlug( product.product_slug ) ) {
+			numberOfMailboxes = product.volume ?? null;
+		}
+
+		if ( isGoogleWorkspaceProductSlug( product.product_slug ) || isTitanMail( product ) ) {
+			numberOfMailboxes = product.current_quantity ?? null;
+		}
+
+		if ( numberOfMailboxes === null ) {
+			return;
+		}
+
+		return (
+			<LineItemMeta>
+				{ translate(
+					'%(numberOfMailboxes)d mailbox for %(domainName)s',
+					'%(numberOfMailboxes)d mailboxes for %(domainName)s',
+					{
+						args: {
+							numberOfMailboxes,
+							domainName: product.meta,
+						},
+						count: numberOfMailboxes,
+					}
+				) }
+			</LineItemMeta>
+		);
+	}
+
+	let mailboxes = [];
+
+	if (
+		isGoogleWorkspaceProductSlug( product.product_slug ) ||
+		isGSuiteOrExtraLicenseProductSlug( product.product_slug )
+	) {
+		mailboxes = product.extra?.google_apps_users ?? [];
+	}
+
+	if ( isTitanMail( product ) ) {
+		mailboxes = product.extra?.email_users ?? [];
```

Comment:
We need to ensure that `email_users` is correctly defined on the `extra` type definition. (See the general list of TS failures for more.)

Reply:
A quick note that this should be fixed by https://github.com/Automattic/wp-calypso/pull/58761.